ang/civicase/case/details/people-tab/directives/case-details-people-tab.directive.js

```diff
@@ -240,35 +240,58 @@
     /**
      * Unassign the role to a contact
      *
-     * @param {string} role role
+     * @param {object} role role
      */
     $scope.unassignRole = function (role) {
-      CRM.confirm({
-        title: ts('Remove %1', { 1: role.role }),
-        message: ts('Remove %1 as %2?', { 1: role.display_name, 2: role.role })
-      }).on('crmConfirm:yes', function () {
-        var apiCalls = [];
-
-        // when client
-        if (!role.relationship_type_id) {
-          apiCalls = [unassignClientCall(role)];
+      // when client
+      if (!role.relationship_type_id) {
+        CRM.confirm({
+          title: ts('Remove %1', { 1: role.role }),
+          message: ts('Remove %1 as %2?', { 1: role.display_name, 2: role.role })
+        }).on('crmConfirm:yes', function () {
+          var apiCalls = [unassignClientCall(role)];
           getApiParamsToSetRelationshipsAsInactiveWhenClientIsRemoved(role, apiCalls);
-        } else {
-          apiCalls = [unassignRoleCall(role)];
-        }
 
-        apiCalls.push(['Activity', 'create', {
-          case_id: item.id,
-          target_contact_id: role.contact_id,
-          status_id: 'Completed',
-          activity_type_id: role.relationship_type_id ? 'Remove Case Role' : 'Remove Client From Case',
-          subject: ts('%1 removed as %2', { 1: role.display_name, 2: role.role })
-        }]);
+          makeAPICalltoUnassignRole(apiCalls, 'Remove Client From Case', role);
+        });
+      } else {
+        promptForContact(
+          {
+            title: ts('Remove %1', { 1: role.role }),
+            showDescriptionField: false,
+            hideContactField: true,
+            role: role,
+            endDate: {
+              minDate: role.start_date,
+              value: moment().format('YYYY-MM-DD')
+            }
+          },
+          function (contactPromptResult) {
+            var apiCalls = [unassignRoleCall(role, contactPromptResult.endDate)];
 
-        $scope.refresh(apiCalls);
-      });
+            makeAPICalltoUnassignRole(apiCalls, 'Remove Case Role', role);
+          }
+        );
+      }
     };
 
+    /**
+     * @param {object[]} apiCalls list of api calls
+     * @param {string} activityType activity type
+     * @param {object} role role object
+     */
+    function makeAPICalltoUnassignRole (apiCalls, activityType, role) {
```

Comment:
Was thinking this function could have been renamed and made more generic so it accepts an activity subject too. That way it can be used elsewhere to create an activity for the people tab operations or what do you think?

Reply:
There was an existing function `getCreateRoleActivityApiCall`, I have renamed to `getActivityApiCallRelatedToRole` and used the same.